test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -548,4 +555,43 @@ UCS_TEST_P(test_uct_cm_sockaddr, cm_server_reject)
                 (TEST_CM_STATE_SERVER_CONNECTED | TEST_CM_STATE_CLIENT_CONNECTED)));
 }
 
+UCS_TEST_P(test_uct_cm_sockaddr, server_switch_cm)
+{
+    UCS_TEST_MESSAGE << "Testing " << m_listen_addr
+                     << " Interface: " << GetParam()->dev_name;
+
+    m_switch_cm = true;
+
+    ucs_async_context_t *async;
+    uct_worker_h worker_test;
+    ucs_status_t status;
+
+    status = ucs_async_context_create(UCS_ASYNC_MODE_THREAD_SPINLOCK, &async);
+    ASSERT_UCS_OK(status);
+
+    status = uct_worker_create(async, UCS_THREAD_MODE_SINGLE, &worker_test);
+    ASSERT_UCS_OK(status);
+
+    status = uct_cm_open(GetParam()->component, worker_test, &m_cm_test);
```

Comment:
pls use UCS_CREATE_HANDLE

Reply:
@yosefe using a HANDLE for the new cm messes up the destroy flow - 
if the new cm is a member of the class and isn't destroyed in the test itself, it will be destroyed after the 'entity' destroy flow, which will also destroy the async context and result in:

`async 0x1545640 handler 0x1546cf0 [id=14] uct_rdmacm_cm_event_handler() uct_rdmacm_cm_event_handler() not released`

this handler is removed, but later - when the cm handle destroys the new cm.

Reply:
we can reset() the handle to destroy when we want
ucs::handle makes sure there is no leak if exception is thrown